lib/membrane_rtc_engine/engine.ex

```diff
@@ -185,6 +141,38 @@ defmodule Membrane.RTC.Engine do
           display_manager?: boolean()
         ]
 
+  defmodule State do
+    @moduledoc false
+
+    use Bunch.Access
+
+    @enforce_keys [:id, :component_path, :trace_context, :telemetry_label]
+    defstruct @enforce_keys ++
+                [
+                  peers: %{},
+                  endpoints: %{},
+                  pending_subscriptions: [],
+                  pending_peers: %{},
+                  filters: %{},
+                  subscriptions: %{},
+                  display_manager: nil
+                ]
+
+    @type t() :: %__MODULE__{
+            id: String.t(),
+            component_path: String.t(),
+            trace_context: map(),
+            telemetry_label: Membrane.TelemetryMetrics.label(),
+            display_manager: pid() | nil,
+            peers: %{Endpoint.id() => Peer.t()},
+            endpoints: %{Endpoint.id() => Endpoint.t()},
+            filters: map(),
+            subscriptions: %{Endpoint.id() => map()},
+            pending_subscriptions: list(),
+            pending_peers: %{Endpoint.id() => %{peer: Peer.t(), endpoint: Endpoint.t()}}
+          }
+  end
```

Comment:
More detailed specs would be appreciated. I meanr for `subscriptions`, `pending_subscriptions`, and maybe `filters`?  Actually, `filters` seems  to be unused so we can delete them

Reply:
Yeah, it's because I'm wasn't 100% sure what they were, so I left a very generic type because the only thing worse than no documentation is bad documentation.